### PR TITLE
Added a form component to login page. Can login through enter key

### LINKED ISSUE #135: Press enter on Login Screen to log in
**User Story**
As a user
I want to log in quickly
so I can save mouseclicks

**Priority**
[]: High
[]: Medium
[x]: Low

**Describe the solution you want to implement**
PRessing enter in the login screen should log you in

**Describe alternatives you've considered**

**Issue Dependencies**
Does this issue depend on any other issues?

**Additional context**

### PR DESCRIPTION
**Pull request by tyger-kong**
Closes #135 

**Description**

A `form` component has been used to wrap nearly all the components in the page. Also, the `Sign In` button has been given the prop 'type="submit"'. This is to allow the form to be submitted (i.e. button to be clicked) on 'Enter' press

**Testing**
Tested that both clicking the button and pressing enter key has the same effect

**Checklist**

- [x ] I have tested my change
- [x ] Code builds successfully and all tests pass
- [ ] Docs have been added/updated if needed

### CODE REVIEW

**Review by joshiefu — APPROVED** [review]
I approve!

**Comment by ErnWong** [comment]
Is it just me or is the form spamming onSubmit all the time when the sign-in page is open? I'll have a quick sanity check...

**Comment by ErnWong** [comment]
Oh - found the bug!

File: split-webapp/split/src/pages/Login.js [hunk 1]
```diff
@@ -69,94 +69,94 @@ class Login extends Component {
...
-                      variant="filled"
-                      InputProps={{
-                        startAdornment: (
-                          <InputAdornment position="start">
-                            <AccountCircle />
-                          </InputAdornment>
-                        )
-                      }}
-                      onChange={this.handleOnChangeUser}
+        <form onSubmit={this.createDetails()}>
```
Comment (ErnWong) [inline split-webapp/split/src/pages/Login.js]: Bug: It was calling `this.createDetails()` on every re-render.

```suggestion
        <form onSubmit={() => this.createDetails()}>
```

or

```
        <form onSubmit={this.createDetails}>
```

and turn `createDetails` into an arrow function member.

**Review by ErnWong — CHANGES_REQUESTED** [review]

File: split-webapp/split/src/pages/Login.js [hunk 2]
```diff
@@ -23,6 +23,16 @@ class Login extends Component {
     };
   }
 
+  componentDidMount = () => {
+    const inputField = document.getElementById("inner-container-id");
+    inputField.addEventListener("keyup", function(event) {
```
Comment (ErnWong) [inline split-webapp/split/src/pages/Login.js]: Would using the `onKeyUp` prop on the `<Box>` be clearer and safer (similar to the other event handlers)? Or was there a reason why the keyup handler was registered inside `componentDidMount`?

File: split-webapp/split/src/pages/Login.js [hunk 3]
```diff
@@ -23,6 +23,16 @@ class Login extends Component {
...
+      if (event.keyCode === 13) {
+        event.preventDefault();
+        document.getElementById("login-button").click();
```
Comment (ErnWong) [inline split-webapp/split/src/pages/Login.js]: Would using refs be safer over `getElementById`? Or do you reckon it's fine.

File: split-webapp/split/src/pages/Login.js [hunk 4]
```diff
@@ -23,6 +23,16 @@ class Login extends Component {
     };
   }
 
+  componentDidMount = () => {
+    const inputField = document.getElementById("inner-container-id");
+    inputField.addEventListener("keyup", function(event) {
+      if (event.keyCode === 13) {
```
Comment (ErnWong) [inline split-webapp/split/src/pages/Login.js]: Looks like `keyCode` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode). Would using [`event.key === "enter"`](https://caniuse.com/#feat=keyboardevent-key) be a safer choice?

**Review by ErnWong — CHANGES_REQUESTED** [review]

File: split-webapp/split/src/pages/Login.js (hunk 3 above)
Comment (tyger-kong) [reply split-webapp/split/src/pages/Login.js]: It should be fine as no other component has an id field (that i've found)

File: split-webapp/split/src/pages/Login.js (hunk 4 above)
Comment (tyger-kong) [reply split-webapp/split/src/pages/Login.js]: Changes made

File: split-webapp/split/src/pages/Login.js (hunk 2 above)
Comment (tyger-kong) [reply split-webapp/split/src/pages/Login.js]: Changes made to use onKeyUp prop and removed componentDidMount

File: split-webapp/split/src/pages/Login.js (hunk 3 above)
Comment (ErnWong) [reply split-webapp/split/src/pages/Login.js]: Sweet

File: split-webapp/split/src/pages/Login.js (hunk 2 above)
Comment (ErnWong) [reply split-webapp/split/src/pages/Login.js]: Sweet! Looking forward to your changes when pushed.

**Review by ErnWong — APPROVED** [review]
Looks good to me!

**Pull request merged**